internal/xds/rbac/matchers.go

```diff
@@ -0,0 +1,387 @@
+/*
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rbac
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"regexp"
+
+	v3corepb "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	v3rbacpb "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v3"
+	v3route_componentspb "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	v3matcherpb "github.com/envoyproxy/go-control-plane/envoy/type/matcher/v3"
+	"google.golang.org/grpc/internal/xds/matcher"
+)
+
+// TODO: Do we need errors at every node? Should we have checks on each node (i.e. destinationPort should be within a certain range)
```

Comment:
If something can error, it should be handled. Please take care of this TODO.

Reply:
Ok, I was wondering if port number could error, but it seems like since the proto config for it defines it as 32 bits instead of 16 bits, if it's a 32 bit integer, it's a valid config option, even though it's theoretically an invalid port number.

Reply:
Deleted this TODO.